### PR TITLE
fix: invalid variable access

### PR DESCRIPTION
**Pull request by Mike325**
Missing M variable declaration
Use local variables by_(prio,ft,key) instead of the module ones

### CODE REVIEW

**Comment by L3MON4D3** [comment]
Whoops again I guess 😅

File: lua/luasnip/session/snippet_collection.lua [hunk 1]
```diff
@@ -8,6 +8,10 @@ M = {}
 --		ft2 = {...}
 --	}
 -- }
+local M = {
```
Comment (L3MON4D3) [inline lua/luasnip/session/snippet_collection.lua:11]: Could you also remove the other `M`?

Comment (Mike325) [reply lua/luasnip/session/snippet_collection.lua:9]: sure ! I didn't see it 😅

**Comment by L3MON4D3** [comment]
Okay, LGTM, thank you very much :)

**Pull request merged**